app/views.py

```diff
@@ -1765,37 +1771,61 @@ def dyndns_update():
-            if result['status'] == 'ok':
-                history = History(msg='DynDNS update: updated record {0} in zone {1}, it changed from {2} to {3}'.format(hostname,domain.name,oldip,myip), detail=str(result), created_by=current_user.username)
+            if isinstance(ip, (ipaddress.IPv4Address, ipaddress.IPv6Address)):
+                return [ip]
+        return []
+
+    myip_addr = []
+    if myip:
+        for address in myip.split(','):
+            myip_addr += validate_address(address)
+
+    remote_addr = validate_address(request.headers.get('X-Forwarded-For', request.remote_addr).split(', ')[:1])
+
+    response='nochg'
+    for ip in myip_addr or remote_addr:
+        if isinstance(ip, ipaddress.IPv4Address):
+            rtype='A'
+        else:
+            rtype='AAAA'
+
+        r = Record(name=hostname,type=rtype)
+        # check if the user requested record exists within this domain
+        if r.exists(domain.name) and r.is_allowed_edit():
+            if r.data == str(ip):
+                # record content did not change, return 'nochg'
+                history = History(msg="DynDNS update: attempted update of {0} but record did not change".format(hostname), created_by=current_user.username)
                 history.add()
-                return render_template('dyndns.html', response='good'), 200
             else:
-                return render_template('dyndns.html', response='911'), 200
-    elif r.is_allowed_edit():
-        ondemand_creation = DomainSetting.query.filter(DomainSetting.domain == domain).filter(DomainSetting.setting == 'create_via_dyndns').first()
-        if (ondemand_creation != None) and (strtobool(ondemand_creation.value) == True):
-            record = Record(name=hostname,type='A',data=myip,status=False,ttl=3600)
-            result = record.add(domain.name)
-            if result['status'] == 'ok':
-                history = History(msg='DynDNS update: created record {0} in zone {1}, it now represents {2}'.format(hostname,domain.name,myip), detail=str(result), created_by=current_user.username)
-                history.add()
-                return render_template('dyndns.html', response='good'), 200
+                oldip = r.data
+                result = r.update(domain.name, str(ip))
+                if result['status'] == 'ok':
+                    history = History(msg='DynDNS update: updated {0} record {1} in zone {2}, it changed from {3} to {4}'.format(rtype,hostname,domain.name,oldip,str(ip)), detail=str(result), created_by=current_user.username)
+                    history.add()
+                    response='good'
+                else:
+                    response='911'
+                    break
+        elif r.is_allowed_edit():
+            ondemand_creation = DomainSetting.query.filter(DomainSetting.domain == domain).filter(DomainSetting.setting == 'create_via_dyndns').first()
+            if (ondemand_creation != None) and (strtobool(ondemand_creation.value) == True):
```

Comment:
This leads to an exception:
```
[ERROR] Exception on /nic/update [GET]
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "./app/views.py", line 94, in decorated_function
    return f(*args, **kwargs)
  File "./app/views.py", line 1817, in dyndns_update
    if (ondemand_creation != None) and (strtobool(ondemand_creation.value) == True):
  File "./app/models.py", line 734, in __eq__
    return self.setting == other.setting
AttributeError: 'NoneType' object has no attribute 'setting'
```
This is triggered because before the resource type was 'None'. When init the resource type with ipv4 or ipv6, the equality [operator](https://github.com/ngoduykhanh/PowerDNS-Admin/blob/master/app/models.py#L734) triggers an exception as shown above. 

To prevent this you can check with `is not`:

```suggestion
            if (ondemand_creation is not None) and (strtobool(ondemand_creation.value) == True):
```

I will also add a pull request to fix this in the DomainSettings class

Reply:
Merged this PR to solve the issue.